### PR TITLE
PsP reverting back to 1.6.2

### PR DESCRIPTION
**Pull request by saimarathore**
PsP reverting back to 1.6.2

### CODE REVIEW

File: src/applications/PseudoProgressionEstimator.cpp [hunk 1]
```diff
@@ -803,10 +795,8 @@ VariableSizeMatrixType PseudoProgressionEstimator::LoadPseudoProgressionTestingD
     std::cout << "Loading Perfusion Image: " << sid << std::endl;
     std::map<CAPTK::ImageModalityType, std::string> currentsubject = testingsubjects[sid];
     ImageType::Pointer LabelImagePointer = ReadNiftiImage<ImageType>(static_cast<std::string>(currentsubject[CAPTK::ImageModalityType::IMAGE_TYPE_SEG]));
-    //NiftiDataManager m_obj;
-    //auto perfImagePointerNifti = m_obj.Read4DNiftiImage(static_cast<std::string>(currentsubject[CAPTK::ImageModalityType::IMAGE_TYPE_PERFUSION]));
-    using ImageTypePerfusion = itk::Image< float, 4 >;
-    auto perfImagePointerNifti = cbica::ReadImage< ImageTypePerfusion >(static_cast<std::string>(currentsubject[CAPTK::ImageModalityType::IMAGE_TYPE_PERFUSION]));
+    NiftiDataManager m_obj;
+    auto perfImagePointerNifti = m_obj.Read4DNiftiImage(static_cast<std::string>(currentsubject[CAPTK::ImageModalityType::IMAGE_TYPE_PERFUSION]));
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cpp:799]: can cbica::readimage read a 4D image?

File: src/applications/PseudoProgressionEstimator.cpp [hunk 2]
```diff
@@ -1553,13 +1535,13 @@ PerfusionMapType PseudoProgressionEstimator::CombineAndCalculatePerfusionPCA(Per
     for (unsigned int i = 0; i < Features.Rows(); i++)
     {
       VectorDouble oneVector;
-      for (unsigned int j = 0; j < Features.Cols(); j++)
+      for (unsigned int j = 0; j < 45; j++)
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cpp:1538]: what is 45? Please add some comment.

File: src/applications/PseudoProgressionEstimator.cpp [hunk 3]
```diff
@@ -1595,12 +1577,11 @@ PerfusionMapType PseudoProgressionEstimator::CombineAndCalculatePerfusionPCAForT
...
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cpp:1580]: Comment about 45

File: src/applications/PseudoProgressionEstimator.cpp [hunk 4]
```diff
@@ -2054,170 +2060,176 @@ VectorVectorDouble PseudoProgressionEstimator::CombineAllThePerfusionFeaures(Vec
...
   }
 
+
+
+
   VariableSizeMatrixType AllPCAs;
   VariableSizeMatrixType AllMeans;
-  AllPCAs.SetSize(23 * 20, 20);
-  AllMeans.SetSize(23, 20);
+  AllPCAs.SetSize(23 * 255, 255);
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cpp:2110]: what are the magic numbers? Please add some comments

File: src/applications/PseudoProgressionEstimator.cpp [hunk 5]
```diff
@@ -2632,15 +2653,20 @@ void PseudoProgressionEstimator::ReadAllTheModelParameters(std::string modeldire
...
   for (unsigned int i = 0; i < dataMatrix.size(); i++)
     Mean_PERF[i] = dataMatrix(0, i);
 
+
+
+
+
   //-------------others related data reading------------------
-  int PCA_Others_Size = 20;
+  int PCA_Others_Size = 255;
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cpp:2669]: comment about 255

File: src/applications/PseudoProgressionEstimator.cpp [hunk 6]
```diff
@@ -2694,188 +2720,140 @@ void PseudoProgressionEstimator::ReadAllTheModelParameters(std::string modeldire
   Mean_PC10.SetSize(PCA_Others_Size);
 
   int start_counter = 0;
-  int end_counter = 19;
+  int end_counter = 254;
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cpp:2723]: what is 254?

File: src/applications/PseudoProgressionEstimator.cxx [hunk 7]
```diff
@@ -157,20 +156,16 @@ int SurvivalPredictionOnExistingModel(const std::string modeldirectory,
...
-    PseudoProgressionEstimator objPseudoProgressionEstimator;
-    bool result = objPseudoProgressionEstimator.PseudoProgressionEstimateOnExistingModel(QualifiedSubjects, modeldirectory, inputdirectory, outputdirectory, true, true, true, true);
-    if(result==true)
-      std::cout<<"Results saved in the output folder."<<std::endl;
-  }
-
+  std::cout << "Number of subjects with required input: " << QualifiedSubjects.size() << std::endl;
+  PseudoProgressionEstimator objPseudoProgressionEstimator;
+  bool data = objPseudoProgressionEstimator.PseudoProgressionEstimateOnExistingModel(QualifiedSubjects, modeldirectory, inputdirectory, outputdirectory, true, true, true, true);
+  //for (unsigned int subjectID = 0; subjectID < QualifiedSubjects.size(); subjectID++)
```
Comment (ashishsingh18) [inline src/applications/PseudoProgressionEstimator.cxx:162]: Please remove commented code or add some comments

File: src/applications/common_includes/FeatureReductionClass.cpp [hunk 8]
```diff
@@ -1143,11 +2603,230 @@ vtkSmartPointer< vtkTable >  FeatureReductionClass::GetDiscerningPerfusionTimePo
   }
   catch (const std::exception& e1)
   {
-    std::cerr << e1.what() << "\n";
   }
+
   return projectedDatasetTable;
 }
 
+
```
Comment (ashishsingh18) [inline src/applications/common_includes/FeatureReductionClass.cpp:2611]: Please add some documentation about commented code

File: src/applications/PseudoProgressionEstimator.cpp [hunk 9]
```diff
@@ -1,4 +1,5 @@
 #include "PseudoProgressionEstimator.h"
+#include "PseudoProgressionEstimator.h"
```
[the remark this answers is not in the document]
Comment (sarthakpati) [reply src/applications/PseudoProgressionEstimator.cpp:2]: Done.

File: src/applications/PseudoProgressionEstimator.cpp (hunk 1 above)
Comment (sarthakpati) [reply src/applications/PseudoProgressionEstimator.cpp:799]: Yes, and changed.

**Pull request merged**